feat(gnovm): improve interface/func/struct type `String` methods

thehowl opened the pull request:
- only show the types in the parameters and results for functions, matching go's reflect.Type.String
- match field list to go's

extracted from #3847.

This blocks https://github.com/gnolang/gno/pull/3847 because for simplicity, we use the type String() methods for syntaxes like `%#v` and `%T` which show the type. Making these match Go's avoids us modifying some types, and I think it's a nice addition in making us more consistent with Go.

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: thehowl/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

mvertes reviewed (approved):
Looks good to me. Not blocking: I would prefer to comply with the `go fmt` rules when printing `struct` or `interface` on single line (see inline comments).

thehowl commented:
@mvertes I noticed that Go error messages don't actually put a space between the `interface` / `struct` and the respective bracket, in general, so I removed the bracket to revert to the previous state more generally.

I think having error messages more consistent is more important, so I reverted it back.

ltzmaxwell reviewed (approved):
LTGM.
